src/interfaces/assistants_web/src/components/UI/FileViewer.tsx

```diff
@@ -0,0 +1,35 @@
+import { Markdown } from '@/components/Markdown';
+import { Icon } from '@/components/UI/Icon';
+import { Spinner } from '@/components/UI/Spinner';
+import { Text } from '@/components/UI/Text';
+import { useFile } from '@/hooks';
+
+type Props = {
+  fileId: string;
+  agentId?: string;
+  conversationId?: string;
+};
+
+export const FileViewer: React.FC<Props> = ({ fileId, agentId, conversationId }) => {
+  const { data: file, isLoading } = useFile({ fileId, agentId, conversationId });
+
+  if (isLoading) {
+    return <Spinner />;
+  }
+
+  return (
+    <div className="space-y-2.5">
+      <header className="flex items-center gap-2">
+        <div className="rounded bg-mushroom-800 p-2.5 dark:bg-volcanic-150">
+          <Icon name="file" />
+        </div>
+        <Text styleAs="p-lg" className="truncate">
+          {file!.file_name}
+        </Text>
+      </header>
+      <article className="max-h-96 overflow-y-auto">
+        <Markdown className="font-variable" text={file!.file_content} />
```

Comment:
Hopefully it will never come up, but I think we should be handling the cases where `file` is `undefined` - most likely if the file is being loaded over a slow connection or if something really strange happens and the request for the file fails.

For a default value for the title, we could pass in a value in `handleOpenFile` in the `ConversationPanel`. Default content could be as simple as the string "Loading...". If there's an error, then we could put an error message into the `Markdown` component.

Reply:
For slow internet connection we have a spinner. Error handling has been added.